src/java/com/phenix/pct/PCTASBroker.java

```diff
@@ -0,0 +1,401 @@
+                bw.println("autoStart=" + (autoStart ? "1" : "0"));
+                if (server != null) {
+                    if (!"".equals(server.getPropath())) {
+                        bw.print("PROPATH=");
+                        bw.println(server.getPropath());
+                    }
+
+                    // TODO Erm, this is crap... I should use something else to handle correctly quotes in command line
+                    bw.print("srvrStartupParam=");
+                    for (String str : server.getCmdLineParameters()) {
+                        bw.print(str);
+                        bw.print(' ');
+                    }
+
+                    bw.println("");
+                }
+
+                if (portNumber != -1)
+                    bw.println("portNumber=" + portNumber);
+                if (brokerLogFile != null)
+                    bw.println("brokerLogFile=" + brokerLogFile);
+                if (brokerLogLevel != -1)
+                    bw.println("brkrLoggingLevel=" + brokerLogLevel);
+                bw.println("brkrLogAppend=" + (brokerLogFileAppend ? "1" : "0"));
+                if (serverLogFile != null)
+                    bw.println("srvrLogFile=" + serverLogFile);
+                if (serverLogLevel != -1)
+                    bw.println("srvrLoggingLevel=" + serverLogLevel);
+                bw.println("srvrLogAppend=" + (serverLogFileAppend ? "1" : "0"));
+                if (workDir != null)
+                    bw.println("workDir=" + workDir);
+                if (initialPool != -1)
+                    bw.println("initialSrvrInstance=" + initialPool);
+                if (minPool != -1)
+                    bw.println("minSrvrInstance=" + minPool);
+                if (maxPool != -1)
+                    bw.println("maxSrvrInstance=" + maxPool);
+                if ((server != null) && (server.getActivateProc() != null)) {
+                    bw.println("srvrActivateProc=" + server.getActivateProc());
+                }
+                if ((server != null) && (server.getDeactivateProc() != null)) {
+                    bw.println("srvrDeactivateProc=" + server.getDeactivateProc());
+                }
+                if ((server != null) && (server.getConnectProc() != null)) {
+                    bw.println("srvrConnectProc=" + server.getConnectProc());
+                }
+                if ((server != null) && (server.getDisconnectProc() != null)) {
+                    bw.println("srvrDisconnProc=" + server.getDisconnectProc());
+                }
+                if ((server != null) && (server.getStartupProc() != null)) {
+                    bw.println("srvrStartupProc=" + server.getStartupProc());
+                }
+                if ((server != null) && (server.getShutdownProc() != null)) {
+                    bw.println("srvrShutdownProc=" + server.getShutdownProc());
+                }
+                if (uid.equalsIgnoreCase("auto")) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Move the "auto" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1132)